packages/language-server/src/plugins/typescript/astro2tsx.ts

```diff
@@ -27,13 +28,19 @@ export default function(content: string): Astro2TSXResult {
     code: ''
   };
 
-  // Replace frontmatter marks with comments
-  let raw = content
-    // Handle case where semicolons is not used in the frontmatter section
-    .replace(/((?!^)(?<!;)\n)(---)/g, (_whole, start, _dashes) => {
-      return start + ';' + '//';
-    })
-    .replace(/---/g, '///')
+	const astroDocument = parseAstro(content)
```

Comment:
Not sure if using `parseAstro` is allowed here. I first tried to do this by passing `this.doc.astro` to `astro2tsx` in `DocumentSnapshot.ts` but sometimes it returned undefined and it made the extension crash. 

I figure that this cause a double parse of the file but I wasn't sure how to do otherwise